vivisect/base.py

```diff
@@ -537,6 +548,16 @@ def _handleAUTOANALFIN(self, einfo):
         '''
         pass
 
+    def _handleWRITEMEM(self, einfo):
+        '''
+        Handle permanent writes to a memory map after initialization
+        (fname, off, bytez, supv) where supv is supervisor mode...
+        '''
+        fname, off, bytez, supv = einfo
```

Comment:
why was supv included in this change def? Seems unused and unnecessary. Come to think of it, could fname/off be replaced by the va (for perf) at the time of the change since they are discrete and resolved at the time?

Reply:
we typically want envi.MemoryObject to honor a memory map's permissions, unless "we're acting as the kernel" or some debugging like thing.  that's the foundation of Emulators and VivWorkspaces alike.

we need to be able to modify memory locations that may or may not be writeable, and simply not care.

fname/off is intentionally left in there, since we (you and i, ages past) were working toward making things much more relocatable.  i still see a great deal of value in that.

Reply:
Priv is not used, the API always assumes the admin perms. Assuming those perms were checked before issuing the change request, I think we would always want to assume admin privs in the handler.

We should chat about fname/offs because i think this being a resolved VA at the time doesn't conflict with making things relocatable, and solves a potential future foot-gun issue with non-file backed memory maps that would be very useful for unpackers.

Reply:
i'm pretty sure that only the VivWorkspace api assumes admin perms, and only for this writing.  currently, the api assumes normal perms.  reading from a non-read map throws a segv, iinm

let's have the discussion about VA vs. fname/offs, and where.  imo, if any resolution is to take place, it should be *before* firing the handler, so that the fname/offset is stored in the event chain.  i'd like to move away from *any* VA's being included in the event stream, and that includes VaSet data.

Reply:
addressed per our discussion